### DISCUSSION TITLE
Gadgetbridge integration

### DISCUSSION THREAD

**Discussion by Nb-tst** [j-kaltes/Juggluco#33] [Ideas]
First af all, thanks for this useful app and for the GB integration.

Currently ( ver 4.17.4 ) Juggluco uses min and max temps to show current bg, current temp for delta and weather condition for trends. This leaves humidity and wind values unused.
Could you set them to hh:mm of last reading? Also, could you set a specific weather code for the case of a generic error/ signal lost?
I think that this two tweaks could be useful in order not to have misreadings of the actual situation ( when the bg value is not updated for some reason and the displayed one is not the actual one ). Another safety layer would be to have a clear BT not connected icon, but this has to be done in the watchface.
This change does not impact any other feature of the current integration, as far as I know.

Another change you might consider is to calculate trend value over a few points, as the minute change is easily affected by every noise component making the value not indicative. Just as an example, to lessen the noise component could be considered a X minutes span and used a weighted average that gives more weight to the last value.
Eg for a 3 minutes span:
Δ = ( t6+t5+2*t4)/4 - ( t3 + t2 + 2*t1 )/4

**Comment by j-kaltes** [comment]
The trend (direction of change) value is the same as displayed in Juggluco itself and it is **not** determined by only the last value as you can verify yourself.
weatherSpec specification send to GadgetBridge contains a timestamp giving the number of seconds since 00:00 1 January 1970 GMT.  I don't know what happens with this timestamp.
I could add the time to the location, but on miband 4 it will not fit on the screen without scrolling. For other variables I would need to know what watch faces do with it. When I put the hour in currentHumidity and the minutes in windDirection, can that than be used in some way?

**Reply by Nb-tst** [reply]
As far as I know, miband 4 can only display temps on the main watchface. Miband 5- 6 - 7 and Amazfit Band 5 can also show humidity and wind on the watchface ( miBand 6-7 show also AQI which should be some air quality index not yet implemented in GB weatherspec ). I don't know about other huami watches.

The best way to show time of last reading would be to use the sunrise/sunset time entries, which can be displayed on the main watchface.
However, although they are usually provided by the weather app, for some "historical reason" of the GB project  that I did not understand they are not managed by weatherspec, but acquired by the app itself until a rework of the weather protocol:
https://github.com/Freeyourgadget/Gadgetbridge/blob/e1cccd69533ce08c8e48280b99423465b5db4af0/app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/huami/HuamiSupport.java#L3195

I don't know if and how the timestamp is now used.

**Comment by Nb-tst** [comment]
Here is an example of a watchface for a Mi Band 5 with humidity and wind used to show hh:mm. I used an Artem watchface as base.

This watchface is a proof of concept and so **not** yet suitable for daily use.
![jhgfjhfg](https://github.com/j-kaltes/Juggluco/assets/82938674/7dcce0cd-44df-4dc4-ad59-dc7e9fb6f4cf)


In fact arrows are not correctly mapped to weather conditions, so showed trend arrows are wrong, and this combination of wind/humidity/temps might not work on all models.
Also, there are some unused images in it coming from other previous project that would have used some values to show a kind of graph ( it might come back in the future if GB implements sunset/sunrise from weather, leaving some spare digits to be used ).

For a simple watchface editor see here:
https://watchface-web-editor.vercel.app/

To upload the watchface, just click on it in android and open it with Gadgetbridge.
[temp.zip](https://github.com/j-kaltes/Juggluco/files/11529543/temp.zip)

**Reply by j-kaltes** [reply]
<p style="line-height: 100%; margin-bottom: 0in">
This here is an <a href="https://drive.google.com/file/d/1a1EK1bP6P07kl2YkV8gTuN6B33mjybuf/view?usp=share_link">APK (second version)</a>.
I have added the currentHumidity for hour and windDirection for minute. These are all the assignments:
</p><p>
currentCondition
=location             = string containing Arrow + glucose value 
</p>
<p style="line-height: 100%; margin-bottom: 0in">timestamp           
= (int) (timmsec/1000L);</p>
<p style="line-height: 100%; margin-bottom: 0in">currentConditionCode=libreweathercode;</p>
<p style="line-height: 100%; margin-bottom: 0in">windSpeed=rateofChange;</p>
<p style="line-height: 100%; margin-bottom: 0in">currentHumidity=
hour</p>
<p style="line-height: 100%; margin-bottom: 0in">windDirection=minute</p>
<p style="line-height: 100%; margin-bottom: 0in">currentTemp=rateofChange*10;</p>
<p style="line-height: 100%; margin-bottom: 0in">MMOL/L:</p>
<p style="line-height: 100%; margin-bottom: 0in">todayMaxTemp= number
before point</p>
<p style="line-height: 100%; margin-bottom: 0in">todayMinTemp=number
after point</p>
<p style="line-height: 100%; margin-bottom: 0in">MG/DL:</p>
<p style="line-height: 100%; margin-bottom: 0in">todayMaxTemp= all
but lower digit</p>
<p style="line-height: 100%; margin-bottom: 0in">todayMinTemp=lower
digit;</p>
<p style="line-height: 100%; margin-bottom: 0in"><br/>

</p>
<p style="line-height: 100%; margin-bottom: 0in">libre weather code:</p>


Libre arrow | Undetermined | Falling_Quickly | Falling | Steady | Rising | Rising_Quickly
-- | -- | -- | -- | -- | -- | --
Number | 701 | 602 | 600 | 800 | 500 | 502
Meaning | mist | Heavy snow | snow | clear | rain | heavy rain


<p style="line-height: 100%; margin-bottom: 0in"><br/>

</p>
<p style="line-height: 100%; margin-bottom: 0in"><br/>

</p>
</body>
</html>�

**Comment by Nb-tst** [comment]
In my case ( amazfitband5, which is a clone of Mi Band 5 )  the watchface can only show wind speed adn not wind direction.
Also. windspeed is converted to a 0-12 scale ( beaufort scale ) by Gadgetbridge itself, so the app has to be recompiled in order to show the whole range.
This is another test watchface with trend arrow and undetermined symbol ( i put a simple ! ) according to your specs:
[watchface_conditions.zip](https://github.com/j-kaltes/Juggluco/files/11557185/watchface_conditions.zip)

I will add even a miband 4 and 6 test watchface in case someone wanted to give a try.

Also I'm planning to build a Gadgetbridge apk with this two customization:
- Full range windspeed
-  Get sunset/sunrise times from your values in order to have a timestamp display in its proper format, and other digits for better purposes.

**Comment by j-kaltes** [comment]
I could put the minutes in currentTemp.

**Reply by Nb-tst** [reply]
The current configuration could be the right one for a broader compatibility.
In fact, mi band 4 can only show max and min temp besides weather icon, and using a temperature for trends it's the better option as it can have negative values. So that could be right for the core info.
Aiming only at the compatibility of band 5 and later, unless using a customized version of GB, this could be the best configuration:
BG value: two digits from humidity ( it has range 0-255 ) and one from windspeed ( with this scale: )
https://github.com/Freeyourgadget/Gadgetbridge/blob/e1cccd69533ce08c8e48280b99423465b5db4af0/app/src/main/java/nodomain/freeyourgadget/gadgetbridge/model/WeatherSpec.java#L60
Trend: current temp
HHmm of last reading: min and max temp, setting values <10 as negative so to use the minus as a zero in the watchface. Otherwise times like 01:09 would display as 1: 9.

Of course, using a customized version of gadgetbridge ( or the new weather implementation coming in the future ) would be a lot better.

**Reply by j-kaltes** [reply]
windSpeed is float; is that displayed the right way?
The watchfaces for MI band 4 that claim to display weather information, don't do so in reality. https://amazfitwatchfaces.com/search/mi-band-4/tags/weather?page=2

**Reply by Nb-tst** [reply]
> windSpeed is float; is that displayed the right way? 

Last time I dis some test it  worked

>The watchfaces for MI band 4 that claim to display weather information, don't do so in reality. https://amazfitwatchfaces.com/search/mi-band-4/tags/weather?page=2

Seems they show  min/max temp and weather condition icon. 
https://amazfitwatchfaces.com/mi-band-4/view/18840

I don't have a mi band 4 now but I can get one to give a try.

**Reply by j-kaltes** [reply]
Isn't it possible to put hour in windspeed, so that time is given in a 12 hour clock.  Giving windspeed the values (depending on the hour from 0 to 11):  2, 6, 12, 20, 29, 39, 50, 62, 75, 89, 103, 118.
Minutes can then be put into currentHumidity.

**Reply by Nb-tst** [reply]
It is. Time will be shown on a 12h format, but is doable. For the zeros, I could try to arrange them in the background of the watchface, so they're displayed only when there is a single digit value.

**Reply by j-kaltes** [reply]
This an apk with         
```
float[] speeds= {2, 6, 12, 20, 29, 39, 50, 62, 75, 89, 103, 118,150};
weatherSpec.currentHumidity= cal.get(Calendar.MINUTE);
weatherSpec.windSpeed=speeds[cal.get(Calendar.HOUR)];
```
https://drive.google.com/file/d/1ozJLNSWN9gFgF0y7tllioBNlT6VnUCWd/view?usp=share_link

**Reply by Nb-tst** [reply]
Working as expected so far. For hours, it shows a 0 for 12. I noticed that in the gadgetbridge code there are only 12 values instead of 13 as it should be for a 0-12 range, so there can't be a fix outside from gadgetbridge itself.

**Reply by j-kaltes** [reply]
Does 12 happen in a 12 hour clock or is it 0? I could at least make it 0 by adding module 12.

**Reply by j-kaltes** [reply]
Phone version with certain 0-11 hour clock: https://drive.google.com/file/d/1m_GX9GcaHv-rNA8OajX4ZYFbLpUrY7EY/view?usp=share_link

**Reply by Nb-tst** [reply]
> Does 12 happen in a 12 hour clock or is it 0? I could at least make it 0 by adding module 12.

12:mm was displayed as 0:mm, standard 24h clock

**Comment by Nb-tst** [comment]
Yet another test watchface for Mi Band5 -Amazfit Band 5
Trend arrows should work as expected, not sure about the generic error.
It has windspeed: humidity for last reading time.
I put two zeroes on the background, so that time is correctly-sh shown ( 12 hour format, 12 hour shown as 00 ) except for maybe some minor digits misalignement.
[watchface_beta1.zip](https://github.com/j-kaltes/Juggluco/files/11612025/watchface_beta1.zip)
It has a lot of spare space, so other elements could be easily added, and with some other tester for mi band 4-6-7 it should be trivial to adapt to those models.

**Comment by Nb-tst** [comment]
Also, I've built some personal versions of GadgetBridge in order to use sunrise/sunset values for a clean hh:mm display leaving other values better purposes.
The goal would be to use digits of wind/humidity and maybe other ( temp?) to display a sort of graph based on the idea of using digits 1-9 as elements of a graph given a proper image mapping. Say that this is a 5, it would represent a central value:
![afa](https://github.com/j-kaltes/Juggluco/assets/82938674/73707ed3-7f44-47f8-bf75-46edf0016625)

having only 9 images to combine, the graph will be limited but still useful. I think that a scheme like:
1 - below 70
2 - 70-89
3 -  90/109
4 - 110/129
5 - 130/149
6 - 150/169
7 - 170/189
8 - 190/219
9 - over 220
mapped to images would be informative.
About values to fill in, I was thinking about averages of 5 minutes chunks for the last 30 minutes ( 6 digits x 5 minutes = 30 minutes span ).
To send values to the version of GB I would publish without breaking the compatibility with the standard version, forecast instances in weatherspec could be used. In fact, just populating forecast of min/max temp for the next two days would be enough, the rest of the translation could be done in GB.
Do you think that this is do-able anytime?

**Reply by j-kaltes** [reply]
Do you want to display not only the last value and an arrow, but a graph with the last points on the watch?
Can't you fill the free space on the watch face with something else? 
Everyone will also have a phone nearby, so if they want to study their past glucose values, they can use their phone.

**Reply by Nb-tst** [reply]
I think that it would be useful to have some info on the general trend and not to have to open the phone to get the the "big picture" of what is happening. At the same time I understand that this may not be worth the effort to add this feature.
Of course the watchface can otherwise be filled with other info beside those related to diabetes.

**Comment by Nb-tst** [comment]
These are two test watchfaces for mi band 4 and mi/amazfit band 5. I use the latter as daily watchface.
Any kind of feedback is welcome from everyone in order to spot issues.
Preview are generated with random numbers, the fourth digit of bg values is covering the trend arrow.
![mb4](https://github.com/j-kaltes/Juggluco/assets/82938674/f540f110-475e-4e0d-afda-32d79269f359)

[watchface_pulse_mb4.zip](https://github.com/j-kaltes/Juggluco/files/11725359/watchface_pulse_mb4.zip)

![mb5](https://github.com/j-kaltes/Juggluco/assets/82938674/78d40c5f-604d-4baa-ba92-3e55cfff5029)

[watchface_pulse_mb5.zip](https://github.com/j-kaltes/Juggluco/files/11725364/watchface_pulse_mb5.zip)

**Comment by Kosmadeo** [comment]
Miband 5 - tried the test watchface (watchface_pulse_mb5.zip), the glucose data sync hour doesn't work, only the minute works. Could you please add this functionality to the final version of Juggluco? I tested it with Juggluco 5.0.0
Thank you in advance ;)

**Reply by Nb-tst** [reply]
Which version of Gadgetbridge are you using?
I tested the  watchface with a Amazfit Band 5 and a Mi band 7 ( Zepp OS ). For some reason the latter always shows an hour more.
See here:
https://codeberg.org/Freeyourgadget/Gadgetbridge/issues/3144
Showing no hour it's weird.

**Comment by Kosmadeo** [comment]
![0164](https://github.com/j-kaltes/Juggluco/assets/137445936/9899d467-37b2-4f5b-92dd-67ef0a42bbfa)

It's my current watchface for Miband 5, I want add to it ifo for glucose data sync.

**Reply by Nb-tst** [reply]
Could you upload the watchface bin?

**Comment by Kosmadeo** [comment]
I just don't know how to use it. I looked at the json file in your dial, there the hour is done like this:

    "Wind": {
      "text": {
        "TopLeftX": 50,
        "TopLeftY": 224,
        "BottomRightX": 99,
        "BottomRightY": 240,
        "Alignment": "Right",
        "SpacingX": 0,
        "SpacingY": 0,
        "ImageIndex": 82,
        "ImagesCount": 10
      },
      "ImageSuffixEN": 45
    }
I am using Gadgetbridge version 0.75.0 (latest). And it doesn't work. Looks like Gadgetbridge doesn't get the info and doesn't pass it on.

I used your original watchface to see if it works.

I'm not a programmer, I use Watch Face for Mi Band 5 & 6 Editor to make watchaces.


I wish this would work, I really like how Gadgetbridge's glucose display works, for me it's super stable and just always works. Suspensions are unlikely to happen, but I would like to know when the last blood glucose reading was.
I am asking for help and thank you in advance.
[Max watchface.zip](https://github.com/j-kaltes/Juggluco/files/11995872/Max.watchface.zip)
![0164](https://github.com/j-kaltes/Juggluco/assets/137445936/c0ede220-ef50-4180-a6ba-c4e81dabe883)

**Reply by Nb-tst** [reply]
Let's try to investigate this.
Does the weather app in the watch report some wind value?
Is it the band firmware updated?

**Reply by Kosmadeo** [reply]
Miband 5 firmware wersion: V 1.0.2.78.
I use Sansung Note 20 5g. Gadgetbridge does not have any weather app configured.

**Reply by Kosmadeo** [reply]
Glucose uploads to the Gadgetbridge work automatically after the Miband 5 is installed and connected, with no additional weather setup.
I am using the latest version of Juggluco and Gadgetbridge.

**Reply by Nb-tst** [reply]
Yes. You should also have some weather app/screen on the band. Does that report any wind value?

**Reply by Kosmadeo** [reply]
Yes, but the wind value displayed on the band via Gadgetbridge is always 0 and it does not change regardless of what the weather app shows.

**Reply by Nb-tst** [reply]
Ok, looks like gadgetbrigde is not sending the wind value then. This is how it looks on my Amazfit Band 5 weather "app":
![image](https://github.com/j-kaltes/Juggluco/assets/82938674/b66c12e3-7a96-4f63-8a90-cde9e275ca9b)

**Reply by Kosmadeo** [reply]
![20230709_210116](https://github.com/j-kaltes/Juggluco/assets/137445936/a15cd78a-b569-40a8-bcb0-0499e507c04d)
![Screenshot_20230709_210133_Weather](https://github.com/j-kaltes/Juggluco/assets/137445936/8be71077-8025-4126-8ee7-675421b3d8f0)

The wind speed is 3; my watchface is always 0

**Reply by Kosmadeo** [reply]
![20230709_210636](https://github.com/j-kaltes/Juggluco/assets/137445936/5105e5d3-66f0-4a9a-aedf-88208039aa41)
To ciekawe, bo tutaj wygląda że jest dobrze, a na tarczy się nie pokazuje. Nie wiem czy to dlatego, że mam tryb 24godzinny ustawiony w MiBandzie?

**Reply by Kosmadeo** [reply]
I changed the time format to 12 hour, it doesn't work.

**Reply by Nb-tst** [reply]
To my knowledge Gadgetbridge works only with some weather apps, you have to check the weather app inside the band and compare that to the watchface. If a wind value is received by the watch and not displayed by the watchface, it's a watchface issue. 
Otherwise it's something else ( competing apps sending different weather info, gadgetbridge issue,  IDK ).

You could also check with this watchface, which has wind value. If this work, we can just copy the json wind section.
*https://amazfitwatchfaces.com/mi-band-5/view/2534*
Nevermind, even this watchface seems to have issues ( see comment section )

**Reply by Nb-tst** [reply]
Saw the new comment, looks like a watchface issue.

**Reply by Kosmadeo** [reply]
From what I've noticed, Gadgetbridge receives the hour of the BG reading as wind, transmits it to Miband - you can see it in the Miband weather section, but it doesn't show up on the watch face. Watchface I'm using yours, no modifications.

**Reply by Nb-tst** [reply]
This is the same watchface where I've changed just the alignment entry and added the CN suffixes. Check if this works:

[watchface_pulse_mb5_v2.zip](https://github.com/j-kaltes/Juggluco/files/11995937/watchface_pulse_mb5_v2.zip)

**Comment by Kosmadeo** [comment]
Doesn't work ;(
![20230709_210116](https://github.com/j-kaltes/Juggluco/assets/137445936/6c177032-97dd-429e-a9bb-5dab21e2a125)
![20230709_210636](https://github.com/j-kaltes/Juggluco/assets/137445936/4f8f83ed-b597-44de-8705-e833fb082b63)

**Reply by Nb-tst** [reply]
Would you try this one just to check if wind can be displayed on your band?
https://amazfitwatchfaces.com/mi-band-5/view/2179

**Comment by Kosmadeo** [comment]
Wind is empty ;(
![20230709_214606](https://github.com/j-kaltes/Juggluco/assets/137445936/2c644456-6145-414a-9f3d-e6d913a21155)
![20230709_214626](https://github.com/j-kaltes/Juggluco/assets/137445936/37a4ff62-1177-4800-9e10-d00f9a3437db)

**Reply by Nb-tst** [reply]
Ok. Looks like this is a known issue, for some versions and firmware at least.
/https://amazfitwatchfaces.com/forum/viewtopic.php?p=9325#p9325
( see comment here )
https://amazfitwatchfaces.com/mi-band-5/view/269

I'm personally testing a watchface with humidity for BG values, current temp for trend, max and min temp for HH:mm
This has the big downside to limit bg values to 254, which is the humidity range in my band,  ( for me is enough informative at the moment, over that value I'm just high and I can always check the phone for the actual value ).
To go beyond that value,  bg value can be divided by ten and a zero can be put in weather condition, so to display 260-270 -280 and so on. It would introduce a max 10 difference with the phone value, but this would be well inside error margins of interstitial testing at those levels ( eg a 10 mg/dl error at 280 mg/dl is negligible ).
If this is preferred, we could test this and ask @j-kaltes if he would like to edit Juggluco accordingly.
This would break compatibility with mi band 4, but given that current solution does not always work I think it would be better to aim for compatibility with latest bands (5-6-7 )

**Reply by j-kaltes** [reply]
What do you want to do with 2.2 mmol/L -  27.8 mmol/L?

**Reply by Nb-tst** [reply]
I don't know, I was focused only to the mg/dl version atm.
I can think about it and propose a solution.

**Reply by Nb-tst** [reply]
Think that the best solution would be to round digits corners, add a little spacing between digits and draw a dot in the background. This is an instant preview with just added spacing and dot in the background, to edit digits corner it would take a little time.
![image](https://github.com/j-kaltes/Juggluco/assets/82938674/5798ba11-09d1-4e68-87f2-79f05ee66448)

**Comment by Nb-tst** [comment]
I made a little recap of the possible options for the watchface disposition. Comments and suggestion are welcome.

![image](https://github.com/j-kaltes/Juggluco/assets/82938674/7ea68f16-7ce3-402c-b3f0-512322ef9951)
![image](https://github.com/j-kaltes/Juggluco/assets/82938674/13139dbb-20ee-43b8-93c1-325a5d8790ff)
![image](https://github.com/j-kaltes/Juggluco/assets/82938674/bc925062-6a4a-4962-9337-09a44a71bc84)
![image](https://github.com/j-kaltes/Juggluco/assets/82938674/26831ac7-bc44-47e5-8df1-599d0b05a669)

**Comment by Nb-tst** [comment]
@Kosmadeo according to this switching the band language to EN would solve the issue.
https://github.com/Johnson070/MiBand-5-watchface-editor/issues/3

**Reply by Nb-tst** [reply]
Tested on Amazfit band 5, switching to the band to a non EN CN language breaks the wind feature on the watchface.

**Comment by Kosmadeo** [comment]
I checked, set the Miband language to English - it works!!! :) thanks a lot ;)

**Comment by Kosmadeo** [comment]
Here is my watch face for MIBand5
![Max watchface](https://github.com/j-kaltes/Juggluco/assets/137445936/42ca569b-42d1-4b10-b8d9-2b9311713bb1)
[Max watchface.zip](https://github.com/j-kaltes/Juggluco/files/12004921/Max.watchface.zip)
![0168](https://github.com/j-kaltes/Juggluco/assets/137445936/8f081d5b-6766-427b-9d67-c753a99c8162)

**Reply by Nb-tst** [reply]
Check if current wind/hour setting is enough for you, it has some limitations because of how wind speed is treated by Gadgetbridge, resulting in limited range ( 0-11/12).
I think it could be better to use the wind as last digit in BG value ( 0-9 is enough for that ) and use a temp for reading hours, so to have a full 0-24 range.
This would break compatibility with MB4, but given that is an old device ( Mi Band 8 is the current one, and you can easily find a 5-6-7 for 20-30 €  ) and that it can't anyway show any reading time, dropping its support would be a minor loss IMO.

**Comment by Kosmadeo** [comment]
Everything seems to be working.
I have one more problem. Miband 5 freezes for a few seconds when updating glucose levels. Is there anything I can do to make Miband run smoothly? It's annoying when you turn it on to check your blood glucose and it doesn't respond for a few seconds.
In the settings of all programs (Juggluco, Gadgetbridge, Xdrip) and bluetooth I have power saving turned off.

**Reply by Nb-tst** [reply]
I don't know, I'm not experiencing any freeze or slowdown.

**Comment by Kosmadeo** [comment]
That's another problem ;) 
Notifications from Juggluco.

Juggluco, after selecting the option: "Glucose level in the status bar", sends the blood glucose level as a notification. I want to use Juggluco alarms on my watch. I set alarms, in Gadgetbridge I added Juggluco as an application from which notifications are displayed and - it does not work as I would like. The bG level notification and the high bG alarm appear in parallel. I don't want BG notifications, just alarms.
I tried to handle it with notification filters in Gadgetbridge, but the filter doesn't work. Whatever word I type and whatever setting I choose, the filtering doesn't work.
I wanted to use Juggluco for glucose alarms on the wristband because LibreLink likes to lose the signal and alarms do not work. Xdrip shows delayed alarms (glucose levels are averaged there).

In the settings of Jouggluco watches, I have only GadgetBridge selected.

I am asking for help here as well.
thank you in advance

**Reply by Nb-tst** [reply]
Yes, notification are not working.
With xDrip, they work ( messages only, no alarms like with watchdrip ).
With latest version of xDrip, if you set 10 minutes averaging ( instead of the 20 minutes default ) and enable one minute values ( engineering mode ) the delay is greatly reduced, so I use that for notification.

**Reply by j-kaltes** [reply]
Did you set Left menu->Watch->Separate?

**Reply by Nb-tst** [reply]
Working with that option enabled.

**Comment by Kosmadeo** [comment]
If in the left menu of Juggluco I select: Watch=>"GadgetBridge" and "Separately"
it does not work. I get Juggluco sugar alerts every minute.


In "Settings" I have "Glucose in status bar" turned on.

**Reply by j-kaltes** [reply]
Turn off Left menu->Watch->Notify.

**Reply by Kosmadeo** [reply]
It's disabled from the start.
 in the left menu of Juggluco I select: Watch=>"GadgetBridge" and "Separately"
And nothing else.

**Reply by j-kaltes** [reply]
And your problem is that you receive every minute a glucose notification on your watch?

**Comment by Kosmadeo** [comment]
Yes, I would like to only see high and low alerts on my watch, not every minute glucose alerts. I tried to use notification filters in Gadgetbridge for this, but they don't work.

**Reply by Nb-tst** [reply]
You can have them *as messages* by setting the alarm and tweaking duration/minutes deactivated entries in ringtone to fit your preferences.

You can't have them *as alarm*, eg vibrating band until you dismiss the alarm on the watch itself, as with watchdrip.
You would probably need to use gadgetbridge intents for that.

**Comment by divest0ber** [comment]
Hello,

i have a problem with my amazfit GTR 2e.
The minimum temp is the Same as current temp.

Is this a known error with juggluco?
With FreeThree and Gadget-Bridge there is no problem.

Thx

**Reply by j-kaltes** [reply]
The meaning of minimal and current temperature send by Juggluco to Gadget Bridge has absolutely nothing to do with the value given by other apps.
Current temperature contains: the rate of change times 10.
The minimal temperature is dependent on the unit:
If mg/dL is used, minimal temperature contains the lower digit of the current glucose. Maximal temperature contains the higher two digits.
If mmol/L is used, minimal temperature contains the glucose value after the point. Maximal temperature contains the glucose value before the point.
As described here:
https://github.com/j-kaltes/Juggluco/discussions/33#discussioncomment-5966065
See also the source of Juggluco:
https://github.com/j-kaltes/Juggluco/blob/2880bbaf61bf8cc1d1e380caf022f12fdb1dc753/Common/src/mobile/java/tk/glucodata/Gadgetbridge.java#L37-L51

**Reply by divest0ber** [reply]
Thank you

My problem is that the minimum temperatur is not the lower digit of my Glucose value. It is the current temperatur and this is not the lower digit of my Glucose value. Maximum temperatur is correct. I use mg/dl.
The location is correct, but i cant use the location with my watch.

Thank you

**Reply by j-kaltes** [reply]
Can you show me some screenshots showing these current temperature and minimal temperature and maximal temperature?
Also send the  saved stream values (Left middle menu->Export->Stream).
It is absolutely impossible that the minimal temperature send by Juggluco contains the rate of change if you use the current version of Juggluco. What version do you use?

**Reply by Nb-tst** [reply]
@divest0ber 
See here, Amazfit GTR 47 had some limitation with temps ( max has to be bigger than min and so on ) , maybe the same issue affects your watch.
I don't know if this happens because of GB ( in that case this could be addressed tweaking its code ) or because of the huami OS.
https://github.com/NightscoutFoundation/xDrip/discussions/2125#discussion-4075797

**Comment by Nb-tst** [comment]
Watchface for Mi Band 6, mostly working, still testing.
![ufyhg](https://github.com/j-kaltes/Juggluco/assets/82938674/8fefa581-2b30-4631-9bbd-a739e4567050)

[watchface6red-m.zip](https://github.com/j-kaltes/Juggluco/files/13440012/watchface6red-m.zip)


Credits for the original watchface:
https://amazfitwatchfaces.com/mi-band-6/view/807

**Reply by Nb-tst** [reply]
Moved heart rate and distance, tweaked digit size for better view, added a zero to hours in the background.
For some reason the battery bar and the alarm icon are not working.
[watchface6red-o.zip](https://github.com/j-kaltes/Juggluco/files/13444958/watchface6red-o.zip)
BT disconnected
![GHbg1](https://github.com/j-kaltes/Juggluco/assets/82938674/5b22aad6-f996-461c-a305-780c5dde1e4e)
BT connected
![GHbg2](https://github.com/j-kaltes/Juggluco/assets/82938674/a9a47774-74f8-4dbd-a989-dccab60bb292)

**Reply by Nb-tst** [reply]
Added zero in background for last reading minutes, added am/pm indicator to main clock, better digit spacing, alarm and battery bar are now displaying in the right way. It should work as daily watchface. Still testing though, feedback always welcome.
![beta6wo](https://github.com/j-kaltes/Juggluco/assets/82938674/f7fc123c-14bc-4765-bb5f-65b828966951)
![beta6w](https://github.com/j-kaltes/Juggluco/assets/82938674/73512659-c51d-450f-a681-ebfac895ffd4)

[watchface6red-beta.zip](https://github.com/j-kaltes/Juggluco/files/13452849/watchface6red-beta.zip)

**Comment by Kosmadeo** [comment]
Miband 7 watchface, with AOD.

Adjusting screen brightness - press the top, bottom of the screen.

[003 Miband 7 zielona tarcza Zielone AOD.zip](https://github.com/j-kaltes/Juggluco/files/13443309/003.Miband.7.zielona.tarcza.Zielone.AOD.zip)
![003 Miband 7 zielona tarcza Zielone AOD](https://github.com/j-kaltes/Juggluco/assets/137445936/f4c814a3-6b02-41c8-9d5e-e976ee2575ae)
![003 Miband 7 zielona tarcza Zielone](https://github.com/j-kaltes/Juggluco/assets/137445936/5448f5a5-bab8-4ec6-ab89-43624862e426)

**Comment by ZenithMonk** [comment]
Hey,
sorry to necro-bump, but did anyone try the Gadgetbridge integration with a different smartwatch (e.g. Pinetime)?
